src/components/TodoList/TodoList.tsx

```diff
@@ -1,224 +1,119 @@
 /* eslint-disable */
 import React from 'react';
+import { useAppSelector } from '../../hooks/hooks';
+import classNames from 'classnames';
+import { useDispatch } from 'react-redux';
+import { currentTodoSlice } from '../../features/currentTodo';
+import { Todo } from '../../types/Todo';
 
 export const TodoList: React.FC = () => {
+  const todos = useAppSelector(state => state.todos);
+  const currentTodo = useAppSelector(state => state.currentTodo);
+  const { query, status } = useAppSelector(state => state.filter);
+  const dispatch = useDispatch();
+
+  const handleSelectTodo = (todo: Todo) => {
+    dispatch(currentTodoSlice.actions.addCurrentTodo(todo));
+  };
+
+  const preparedTodos = (todos: Todo[], query: string, filter: string) => {
+    let preparedTodos = [...todos];
+
+    if (query) {
+      const modifiedQuery = query.toLocaleLowerCase();
+
+      preparedTodos = preparedTodos.filter(todo =>
+        todo.title.toLocaleLowerCase().includes(modifiedQuery),
+      );
+    }
+
+    switch (filter) {
+      case 'all':
+        return preparedTodos;
+      case 'active':
+        return preparedTodos.filter(todo => !todo.completed);
+      case 'completed':
+        return preparedTodos.filter(todo => todo.completed);
+      default:
+        return preparedTodos;
+    }
```

Comment:
Ensure that the `filter` parameter in the `preparedTodos` function matches the expected values ('all', 'active', 'completed') for correct filtering logic. This will ensure that the todos are filtered correctly based on the selected status.